neqo-transport/src/quic_datagrams.rs

```diff
@@ -111,7 +111,7 @@ impl QuicDatagrams {
     ) {
         while let Some(dgram) = self.datagrams.pop_front() {
             let len = dgram.len();
-            if builder.remaining() >= len + 1 {
+            if builder.remaining() > len {
```

Comment:
this "+ 1" was for he frame type. I had a comment, i looks like I have lost it in a iteration.

Please add a comment that more than len bytes is needed because of the frame type.

Reply:
The comment is there on the next line.  I'll revise it slightly.  I found a bug here though.  Opening an issue for you.